Submitted & accepted upstream for a release after 0.54.3:
https://github.com/mesonbuild/meson/pull/7370

From 20cd3bde8ae28909123edf41208ee38788d82bd7 Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@oracle.com>
Date: Mon, 8 Jun 2020 12:15:39 -0700
Subject: [PATCH] unittests: Fix 51 ldflagdedup on Solaris

If the linker doesn't support --export-dynamic, skip test to deduplicate
it, since it always fails finding 0 copies instead of 1.
---
 run_unittests.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/run_unittests.py b/run_unittests.py
index 393e6b31e..0c5b3943c 100755
--- a/run_unittests.py
+++ b/run_unittests.py
@@ -6387,6 +6387,11 @@ class LinuxlikeTests(BasePlatformTests):
         testdir = os.path.join(self.unit_test_dir, '52 ldflagdedup')
         if is_cygwin() or is_osx():
             raise unittest.SkipTest('Not applicable on Cygwin or OSX.')
+        env = get_fake_env()
+        cc = env.detect_c_compiler(MachineChoice.HOST)
+        linker = cc.linker
+        if not linker.export_dynamic_args(env):
+            raise unittest.SkipTest('Not applicable for linkers without --export-dynamic')
         self.init(testdir)
         build_ninja = os.path.join(self.builddir, 'build.ninja')
         max_count = 0
-- 
2.19.3

